src/linear_algebra/free_module_pid.lean

```diff
@@ -79,12 +88,33 @@ begin
   exact (submodule.eq_bot_iff _).mp (hϕ ((finsupp.lapply i).comp b.repr) bot_le) _ ⟨x, hx, rfl⟩
 end
 
+lemma eq_bot_of_generator_maximal_range_eq_zero {N O : submodule R M} (b : basis ι R O)
+  (hNO : N ≤ O)
+  {ϕ : O →ₗ[R] R} (hϕ : ∀ (ψ : O →ₗ[R] R), (ϕ.comp $ of_le hNO).range ≤ (ψ.comp $ of_le hNO).range →
```

Comment:
It seems that you are using this assumption `hϕ` several times, and it looks quite intimidating. Do you think it can be packaged behind a nice and readable name?

Reply:
Informally stated, the condition is that `φ(N)` is maximal among all such images `ψ(N)`. I can't find a definition for "is a maximal element", for example Zorn's lemma is stated in the same `x ≤ y → x = y` way. But for `(φ.comp (of_le hNO)).range` (`φ(N)`) I can definitely make up a name like `φ.submodule_image hNO`.